File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2162.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    int sub[n], res=1, i;
    for(i=0;i<n;i++) scanf("%d", &sub[i]);
    
    if(sub[0]>sub[1]){
        for(i=0;i<n-1;i++){
            if(i%2==0){
                if(sub[i]<=sub[i+1]) res=0;
            } else{
                if(sub[i]>=sub[i+1]) res=0;
            }
        }
    } else if(sub[0]<sub[1]){
        for(i=0;i<n-1;i++){
            if(i%2==0){
                if(sub[i]>=sub[i+1]) res=0;
            }else{
                if(sub[i]<=sub[i+1]) res=0;
            }
        }
    } else res=0;
    
    printf("%d\n", res);
    
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2162.py
n = int(input())

sub = [int(i) for i in input().split()]
res = 1

if(sub[0]>sub[1]):
    for i in range(0, n-1):
        if(i%2==0):
            if(sub[i]<=sub[i+1]): res=0
        else:
            if(sub[i]>=sub[i+1]): res=0

elif(sub[0]<sub[1]):
    for i in range(0, n-1):
        if(i%2==0):
            if(sub[i]>=sub[i+1]): res=0
        else:
            if(sub[i]<=sub[i+1]): res=0
else: res=0

print(res)